docs/source/references/config.md

```diff
@@ -1,18 +1,15 @@
 # Iroha Configuration reference
 
-In this document we provide a reference and detailed descriptions of Iroha's configuration options.
+In this document we provide a reference and detailed descriptions of Iroha's configuration options. Also make sure to read thoroughly an important note[^1] for an explanation of underlying types and default values as this should clear up any confusion about them.
```

Comment:
if the info in this note is so important, it shouldn't be a note. I'd put this info in the main body of the text, not in a footnote

Reply:
The point is, it is a note so that it's present both in the text and linked to by the relevant types, and footnote's the only viable way I know that can do it. I can move the right after the above paragraph, though.

Reply:
can't it be a separate section of the document that we link to?

Reply:
the phrasing of the sentence should be changed in any case:

```suggestion
In this document we provide a reference and detailed descriptions of Iroha's configuration options. The options have different underlying types, which are denoted in code as types wrapped in a single `Option<..>` or in a double `Option<Option<..>>`. For the detailed explanation, please refer to the note at the end of this document[^1].
```

I think this also works:

```suggestion
In this document we provide a reference and detailed descriptions of Iroha's configuration options. The options have different underlying types, which are denoted in code as types wrapped in a single `Option<..>` or in a double `Option<Option<..>>`[^1].
```

Reply:
Agreed, sounds quite harsh when I think about it. Thank you.